core-bundle/src/Image/Studio/ImageResult.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Studio;
+
+use Contao\CoreBundle\Image\ImageFactoryInterface;
+use Contao\CoreBundle\Image\PictureFactoryInterface;
+use Contao\Image\ImageDimensions;
+use Contao\Image\ImageInterface;
+use Contao\Image\PictureConfiguration;
+use Contao\Image\PictureInterface;
+use Psr\Container\ContainerInterface;
+use Webmozart\PathUtil\Path;
+
+class ImageResult
+{
+    /**
+     * @var ContainerInterface
+     */
+    protected $locator;
+
+    /**
+     * @var string|ImageInterface
+     */
+    protected $filePathOrImageInterface;
+
+    /**
+     * @var int|string|array|PictureConfiguration|null
+     */
+    protected $sizeConfiguration;
+
+    /**
+     * Cached picture.
+     *
+     * @var PictureInterface|null
+     */
+    protected $picture;
+
+    /**
+     * Cached image dimensions.
+     *
+     * @var ImageDimensions|null
+     */
+    protected $originalDimensions;
+
+    /**
+     * @param string|ImageInterface                      $filePathOrImage
+     * @param array|PictureConfiguration|int|string|null $sizeConfiguration
+     *
+     * @internal Use the Contao\Image\Studio\Studio factory to get an instance of this class
+     */
+    public function __construct(ContainerInterface $locator, $filePathOrImage, $sizeConfiguration = null)
```

Comment:
Same here. Why not inject the required services or parameters directly?

Reply:
see https://github.com/contao/contao/pull/1753#discussion_r455675117